src/s1reader/s1_burst_slc.py

```diff
@@ -404,7 +404,32 @@ def as_dict(self):
             self_as_dict[key] = val
         return self_as_dict
 
-    def bistatic_delay(self, xstep=1, ystep=1):
+
+    def _steps_to_aranges(self, range_step, az_step):
+        ''' convert range_step (meters) and az_step (seconds) into aranges to
+        generate LUT2ds
+        '''
+        n_range = np.ceil(self.width * self.range_pixel_spacing / range_step).astype(int)
+        range_arange = self.starting_range + np.arange(0, n_range) * range_step
+
+        n_az = np.ceil(self.length * self.azimuth_time_interval / az_step).astype(int)
+        rdrgrid = self.as_isce3_radargrid()
+        az_arange = rdrgrid.sensing_start + np.arange(0, n_az) * az_step
+
+        errs = []
+        if not range_arange:
+            errs.append('range')
+        if not az_arange:
+            errs.append('azimuth')
+        if errs:
+            errs = ', '.join(errs)
+            err_str = f'Cannot build aranges from following step(s): {errs}'
```

Comment:
Might be helpful to indicate why you can't build from the steps? 
Or rather, why would either of these fail? from a `self.width=0`? a failure with `self.as_isce3_radargrid().sensing_start`?

Reply:
`width` and `length` should be ok as they're populated directly from the XML [here](https://github.com/opera-adt/s1-reader/blob/d948dc8215a9310eb7c80a496b0b8735028916e3/src/s1reader/s1_reader.py#L551-L552).

If `length * azimuth_time_interval < az_step` or `width * range_pixel_spacing < range_step`, `n_*` can be zero.

I will add check to ensure `*_step > 0`.